src/slurm_plugin/cluster_event_publisher.py

```diff
@@ -0,0 +1,407 @@
+    ):
+        """Create an instance of ClusterEventPublisher with the standard event publisher."""
+        publiser = ClusterEventPublisher._get_event_publisher(
+            event_logger, cluster_name, node_role, component, instance_id, **global_args
+        )
+        return ClusterEventPublisher(publiser, max_list_size)
+
+    @log_exception(logger, "publish_unhealthy_static_node_events", catch_exception=Exception, raise_on_error=False)
+    def publish_unhealthy_static_node_events(
+        self,
+        unhealthy_static_nodes: List[SlurmNode],
+        nodes_in_replacement: Iterable[str],
+        launched_nodes: Iterable[str],
+        failed_nodes: Dict[str, List[str]],
+    ):
+        """Publish events for unhealthy static nodes."""
+        timestamp = ClusterEventPublisher.timestamp()
+
+        for count, error_detail in self._generate_launch_failure_details(failed_nodes):
+            self.publish_event(
+                logging.WARNING if count else logging.DEBUG,
+                "Number of static nodes that failed to launch a backing instance after node maintenance",
+                event_type="node-launch-failure-count",
+                timestamp=timestamp,
+                detail=error_detail,
+            )
+
+        self.publish_event(
+            logging.DEBUG,
+            "After node maintenance, node failed to launch a backing instance",
+            event_type="node-launch-failure",
+            timestamp=timestamp,
+            event_supplier=self._failed_node_supplier(unhealthy_static_nodes, failed_nodes),
+        )
+
+        self.publish_event(
+            logging.DEBUG,
+            "Number of static nodes failing scheduler health check",
+            event_type="static-node-health-check-failure-count",
+            timestamp=timestamp,
+            event_supplier=self._node_list_and_count_supplier(unhealthy_static_nodes),
+        )
+
+        self.publish_event(
+            logging.DEBUG,
+            "Static node failing scheduler health check",
+            event_type="static-node-health-check-failure",
+            timestamp=timestamp,
+            event_supplier=self._unhealthy_node_supplier(unhealthy_static_nodes),
+        )
+
+        self.publish_event(
+            logging.DEBUG,
+            "Number of instances terminated due to backing unhealthy static nodes",
+            event_type="static-node-instance-terminate-count",
+            timestamp=timestamp,
+            event_supplier=self._terminated_instances_supplier(
+                (node for node in unhealthy_static_nodes if node.instance)
+            ),
+        )
+
+        self.publish_event(
+            logging.DEBUG,
+            "After node maintenance, nodes currently in replacement",
+            event_type="static-nodes-in-replacement-count",
+            timestamp=timestamp,
+            event_supplier=self._node_list_and_count_supplier(nodes_in_replacement),
+        )
+
+        self.publish_event(
```

Comment:
how does an event looks like once written into the .events file? can you provide an example?

Reply:
I've added an example event in comments for each event method. Does that work?

Reply:
perfect! thanks